pkg/models/datasource_cache.go

```diff
@@ -155,10 +157,19 @@ func (ds *DataSource) GetHttpTransport() (*dataSourceTransport, error) {
 		IdleConnTimeout:       90 * time.Second,
 	}
 
+	// Set default next round tripper to the default transport
+	var next http.RoundTripper = transport
```

Comment:
I think this is more idiomatic/simpler:
```suggestion
	next := transport
```

Reply:
Maybe it's just me getting used to the `http` package, but I found it helpful to explicitly specify `transport` as a round tripper, as that what allows us to form the middleware chain, with transport being the base. If that's obvious though, then I'm happy to simplify!

Reply:
I still don't see the point to be honest. If it's useful to be explicit about variable types when declaring them, why don't we do it everywhere?

Reply:
>If it's useful to be explicit about variable types when declaring them, why don't we do it everywhere?
Ah well just like anything, not everything has to be black and white. In some cases it can be useful. Though, like I said this is likely just me getting used to `go`. Will fix

Reply:
Thanks. Yeah, I understand there's leeway depending on the situation, I just don't personally see how it differs from the general case.

Reply:
@wbrowne @marefr Ah, now I see. `http.RoundTripper` is an interface. I'm going to play with the code locally, to get a better idea.

Reply:
I think I would write `next := http.RoundTripper(transport)`.

Reply:
Hah I didn't even realize it was necessary in this case. Yeah `next := http.RoundTripper(transport)` seems reasonable I think if people are cool with that